lib/matplotlib/axis.py

```diff
@@ -1012,6 +1012,97 @@ def set_default_intervals(self):
         # attribute, and the derived code below will check for that
         # and use it if it's available (else just use 0..1)
 
+    def _set_lim(self, v0, v1, alt0=None, alt1=None, *,
+                 emit=True, auto, names=("", "")):
+        """
+        Set view limits.
+
+        This method is a helper for the Axes ``set_xlim``, ``set_ylim``, and
+        ``set_zlim`` methods.  This docstring uses names corresponding to
+        ``set_xlim`` for simplicity.
+
+        *names* is the pair of the names of the first two parameters of the
+        Axes method (e.g., "left" and "right").  They are only used to generate
+        error messages; and can be empty if the limits are known to be valid.
+
+        Other parameters are directly forwarded from the Axes limits setter:
+        *v0*, *v1*, *alt0*, and *alt1* map to *left*, *right*, *xmin*, and
+        *xmax* respectively; *emit* and *auto* are used as is.
+        """
+        v0name, v1name = names  # The value names.
+        name, = [name for name, axis in self.axes._get_axis_map().items()
```

Comment:
This seems a really obscure way to get the name to me.  Is there some reason an axis can't know its "name" when it is created?

Reply:
Axises know their "name", but that's e.g. "r"/"theta" for polar axes whereas here we really want "x"/"y" even for polar.  Perhaps Axises should have multiple name attributes for the various use cases, but that would be a bit hairy too...